### DISCUSSION TITLE
how to passing other params in NuxtLink?

### DISCUSSION THREAD

**Discussion by NMTuan** [nuxt/nuxt#16566] [Questions] [answered]
in nuxt2, i have the route like: /news/_id.vue. I can passing other params like this: `:to={name: 'news-id', params: {id: 333, title: 'test news'}}`.

but in nuxt3, its broken. and i find this:
> 4.1.4 remove passing params in router.push
> https://github.com/vuejs/router/issues/1531
```js
  routes: [
    {
      path: '/',
      name: 'index',
      component: () => import('../views/index.vue'),
      props: true               <----------------  LOOK AT THIS
    },
    ...
```

now, in nuxt3, how can i set it ?

**Comment by jinwoo-web** [comment, marked as answer]
// index.vue
```
useRouter().push({
      name: "message",
      state: {
        message: 'this is message'
      }
})
````
// message.vue
```
const route = useRoute()  
console.log(history.state.message) // 'this is message'
```

see also
https://github.com/vuejs/router/blob/main/packages/router/CHANGELOG.md#414-2022-08-22
